File: 254a.py
inp = open('input.txt','r');
out = open('output.txt','w');
n = int(inp.readline());
a = map(int,inp.readline().split())
d = {}

for i in range(2*n):
	if d.get(a[i]) == None:
		d[a[i]] = [i+1]
	else:
		d[a[i]] += [i+1]
flag = 1;
for i in d.keys():
	if len(d[i])%2 !=0:
		flag = 0
		break;
if flag:
	for i in d.keys():
		for j in range(0,len(d[i]),2):
			out.write(str(d[i][j]) + " " +str(d[i][j+1])+'\n')
else:
	out.write('-1')
out.close()

File: 254a.cpp
#include <bits/stdc++.h>

using namespace std;

int main()
{
	vector <int> v[5000];
	FILE *inp,*out;
	inp = fopen("input.txt","r");
	int n,i,j;
	fscanf(inp,"%d",&n);
	for(i=0;i<2*n;i++)
	{
		int x;
		fscanf(inp,"%d",&x);
		x--;
		v[x].push_back(i+1);
	}

	fclose(inp);

	out  = fopen("output.txt","w");

	for(i=0;i<5000;i++)
	{
		if(v[i].size()%2 != 0)
		{
			fprintf(out,"-1\n");
			fclose(out);
			return 0;
		}
	}

	for(i=0;i<5000;i++)
	{
		for(j=0;j<v[i].size();j+=2)
		{
			fprintf(out, "%d %d\n",v[i][j],v[i][j+1]);
		}
	}
	fclose(out);
	return 0;
}